spec/models/bundle_context_spec.rb

```diff
@@ -23,6 +23,25 @@
     it 'is not valid unless bundle_dir exists on filesystem' do
       expect { bc.bundle_dir = 'does/not/exist' }.to change(bc, :valid?).to(false)
     end
+    it { is_expected.to validate_presence_of(:content_structure) }
+    it { is_expected.to validate_presence_of(:bundle_dir) }
+    it { is_expected.to validate_presence_of(:content_metadata_creation) }
+    describe 'project_name' do
+      it { is_expected.to validate_presence_of(:project_name) }
+      it 'is not valid with chars other than alphanum, hyphen and underscore' do
+        attr_hash[:project_name] = 's p a c e s'
+        expect(bc).not_to be_valid
```

Comment:
I don't think this interaction is doing what you expect.  `attr_hash` and `bc` are decoupled after the `subject` is established.  It would read more cleanly (and work) if you just do assignment on the object like:
```ruby
bc.project_name = 'foo'
```
Ideally, start from a valid object and 
```ruby
expect { bc.project_name = 'bad value' }.to change(bc, :valid?).to(false)
```

Reply:
makes sense, i think...

for my own clarification, does that mean that the first time `attr_hash` is set, it does affect `bc` (i.e. for the test this comment is on), and that for subsequent expectations, the changes to `attr_hash` don't affect `bc`?  or is `bc` already set up before this entire test is even entered, because it's declared using `subject`?  or something else?

anyway, i'd agree the suggested usage is more readable.